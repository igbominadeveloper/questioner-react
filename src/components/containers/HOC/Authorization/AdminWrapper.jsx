import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { withRouter } from 'react-router';

import { getItem } from '../../../../utils/helpers';
const AdminWrapper = ({ component: Component, history, location, ...rest }) => {
  const isAdmin = JSON.parse(getItem('user')).isadmin === 1;
  return (
    <Route
      {...rest}
      render={props =>
        isAdmin ? <Component {...props} /> : <Redirect to="/" />
      }
    />
  );
};

export default withRouter(AdminWrapper);
