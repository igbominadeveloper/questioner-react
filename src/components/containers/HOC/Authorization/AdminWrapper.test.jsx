import React from 'react';
import { shallow } from 'enzyme';

import AdminWrapper from './AdminWrapper';

const component = shallow(<AdminWrapper />);

describe('AdminWrapper', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
