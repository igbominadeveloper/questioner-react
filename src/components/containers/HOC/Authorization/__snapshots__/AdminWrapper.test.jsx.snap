// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminWrapper renders without errors 1`] = `
<Route>
  <Component />
</Route>
`;
